### PR TITLE
Fix _unalignedMaxPartSize for logical partitions (#1250890)

### PR DESCRIPTION
**Pull request by vojtechtrefny**
This fixes two problems: free space inside extended partition was
not recognized and free space outside of extended partition was
considered as valid for resizing.

Signed-off-by: Vojtech Trefny <vtrefny@redhat.com>

### CODE REVIEW

File: blivet/devices/partition.py [hunk 1]
```diff
@@ -755,14 +755,23 @@ def _unalignedMaxPartSize(self):
...
+        sector = self.partedPartition.geometry.end + 1
         try:
             partition = self.partedPartition.disk.getPartitionBySector(sector)
         except _ped.PartitionException:
             pass
         else:
-            if partition.type == parted.PARTITION_FREESPACE:
+            # next partition is free space or 'logical' free space
+            if partition.type in (parted.PARTITION_FREESPACE,
+                                  parted.PARTITION_FREESPACE | parted.PARTITION_LOGICAL):
```
Comment (vpodzime) [inline blivet/devices/partition.py]: I'd consider ``if partition.type & parted.PARTITION_FREESPACE`` better readable, but it could be only me. :)

**Comment by vpodzime** [comment]
Other than the nitpick above, this looks good to me. But I'm sure @dwlehman will suggest you to add some tests ;)

File: blivet/devices/partition.py (hunk 1 above)
Comment (dwlehman) [inline blivet/devices/partition.py]: Seconded.

Comment (vojtechtrefny) [inline blivet/devices/partition.py]: Ok

**Comment by vojtechtrefny** [comment]
Fixed the "if" to be simpler. Not sure about the test, I guess it would require too many Mocks and patches and it would probably test mock not blivet :) I think @KosiehBarter could add some resizing to his blivet tests to test resizing of (logical) partitions.

**Comment by vpodzime** [comment]
> Not sure about the test, I guess it would require too many Mocks and patches and it would probably test mock not blivet :) I think @KosiehBarter could add some resizing to his blivet tests to test resizing of (logical) partitions.

Works for me. Thanks for changing that if!

**Comment by vojtechtrefny** [comment]
Pushed to master and f23-branch.

**Pull request closed without merging**